include/fiction/algorithms/simulation/sidb/random_layout_generator.hpp

```diff
@@ -0,0 +1,178 @@
+     */
+    std::pair<typename Lyt::cell, typename Lyt::cell> coordinate_pair;
+    /**
+     * Number of SiDBs that are placed on the layout.
+     */
+    uint64_t number_of_sidbs = 0;
+    /**
+     * If positively charged SiDBs should be prevented, SiDBs are not placed closer than the minimal_spacing.
+     */
+    bool prevent_positive_charges = true;
+    /**
+     * If positively charged SiDBs should be prevented, SiDBs are not placed closer than this value (2 cells as
+     * Euclidean distance by default).
+     */
+    double minimal_spacing = 2;
+    /**
+     * Maximal number of steps to place the given number of SiDBs.
+     */
+    uint64_t maximal_attempts = 10E6;
+};
+
+/**
+ * Generates a random layout of SiDBs based on the provided parameters.
+ *
+ * @tparam Lyt The layout type.
+ * @param params The parameters for generating the random layout.
+ * @param lyt_skeleton A layout to which random cells are added (useful if you need to add random cells to a given
+ * layout).
+ * @return A randomly generated layout of SiDBs.
+ */
+template <typename Lyt>
+Lyt generate_random_layout(const random_layout_params<Lyt>& params, const Lyt& lyt_skeleton)
+{
+    static_assert(is_cell_level_layout_v<Lyt>, "Lyt is not a cell-level layout");
+
+    const uint64_t number_of_sidbs_of_final_layout = lyt_skeleton.num_cells() + params.number_of_sidbs;
+
+    Lyt lyt{};
+
+    if (lyt_skeleton.num_cells() != 0)
+    {
+        lyt_skeleton.foreach_cell([&lyt, &lyt_skeleton](const auto& cell)
+                                  { lyt.assign_cell_type(cell, lyt_skeleton.get_cell_type(cell)); });
+    }
+
+    bool     successful_generation = false;
+    uint64_t attempt_counter       = 0;
+
+    // Stops if either all SiDBs are placed or the maximum number of attempts were performed.
+    while (lyt.num_cells() < number_of_sidbs_of_final_layout && attempt_counter < params.maximal_attempts)
+    {
+        const auto random_coord = random_coordinate(params.coordinate_pair.first, params.coordinate_pair.second);
+
+        bool constraint_violation_positive_sidbs = false;
+
+        if (params.prevent_positive_charges)
+        {
+            // Checks if the new coordinate is not closer than 2 cells (Euclidean distance) from an already
+            // placed SiDB.
+            lyt.foreach_cell(
+                [&lyt, &random_coord, &constraint_violation_positive_sidbs, &params](const auto& c1)
+                {
+                    if (euclidean_distance<Lyt>(lyt, c1, random_coord) < params.minimal_spacing)
+                    {
+                        constraint_violation_positive_sidbs = true;
+                    }
+                });
+        }
+        // If the constraint that no positive SiDBs occur is satisfied, the SiDB is added to the layout.
+        if (!constraint_violation_positive_sidbs)
+        {
+            lyt.assign_cell_type(random_coord, Lyt::cell_type::NORMAL);
+        }
```

Comment:
This as well?

Reply:
@marcelwa sorry, but I don't understand what you mean? 
Instead of ``if-if``, I could use ``if-else``.